pkg/fqdn/namemanager/gc.go

```diff
@@ -138,16 +137,15 @@ func (n *manager) doGC(ctx context.Context) error {
 
 	metrics.FQDNGarbageCollectorCleanedTotal.Add(float64(len(namesToCleanSlice)))
 	namesCount := len(namesToCleanSlice)
-	// Limit the amount of info level logging to some sane amount
-	if namesCount > 20 {
-		// namesToClean is only used for logging after this so we can reslice it in place
-		namesToCleanSlice = namesToCleanSlice[:20]
+
+	if n.gcCache != nil {
+		n.gcCache.AddMultiple(namesToCleanSlice)
```

Comment:
Would it make sense to pass `GCStart` to `AddMultiple` as the timestamp here?

Reply:
yes, very good point! making the change